pymochad/controller.py

```diff
@@ -54,6 +54,23 @@ def __init__(self, server=None, port=1099):
                 "Unable to connect to server %s" % self.server)
         self.socket.setblocking(0)
 
+    def reconnect(self):
+        """Reconnect when mochad server is restarted/lost connection."""
+        if self.socket:
+            self.socket.close()
+        for addr in socket.getaddrinfo(self.server, self.port):
```

Comment:
This block (to L72) is identical to the connection loop in __init__(). It would be better to have just made that a common private function to avoid the duplication. But, I'll just fix that in a follow up patch though